### PR TITLE
Extend .asBool check

### PR DESCRIPTION
**Pull request by bcosma**

### CODE REVIEW

**Review by ccorcoveanu — DISMISSED** [review]

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by danielailie — DISMISSED** [review]

**Review by ccorcoveanu — APPROVED** [review]

File: erdjs/src/smartcontracts/query.ts [hunk 1]
```diff
@@ -127,9 +127,9 @@ export class ContractReturnData {
         this.asBuffer = Buffer.from(this.asBase64, "base64");
         this.asHex = this.asBuffer.toString("hex");
         this.asNumber = parseInt(this.asHex, 16) || 0;
-        this.asBool = this.asNumber != 0;
         this.asBigInt = new BigNumber(`0x${this.asHex || "00"}`, 16);
         this.asString = this.asBuffer.toString();
+        this.asBool = this.asNumber != 0 && this.asString !== "false" && this.asString !== "" && !this.asBigInt.eq(0, 10);
```
Comment (iulianpascalau) [inline erdjs/src/smartcontracts/query.ts:132]: so if this.asNumber == "no", this.asBool will return True :)

Suggestion to change this to:
 ```TS
this.asBool = this.asNumber != 0 && this.asString == "true" && !this.asBigInt.eq(0, 10);
```

Comment (iulianpascalau) [reply erdjs/src/smartcontracts/query.ts:132]: And maybe we should treat the strings in all case configuration. Something like:
```
this.asString.toLower !== "false"
```

Comment (ccorcoveanu) [reply erdjs/src/smartcontracts/query.ts:132]: As bool should be true for everything truthy, I think "I am a helicopter" should also evaluate to `true`. If we want to go further we should lowercase `asString` so "False" will also evaluate correctly.

Comment (ccorcoveanu) [reply erdjs/src/smartcontracts/query.ts:132]: Maybe we can keep the first implementation, and for strings that should evaluate to false we should have a small helper function, checking an array of the form: ["no", "0", "false"]

**Review by iulianpascalau — APPROVED** [review]

**Review by ccorcoveanu — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Pull request merged**